### PR TITLE
Add FP16 PRelu

### PR DESCRIPTION
**Pull request by juncaipeng**
### PR types

Others
### PR changes

OPs
### Describe

Add FP16 PRelu

### CODE REVIEW

File: python/paddle/fluid/tests/unittests/test_prelu_op.py [hunk 1]
```diff
@@ -270,6 +275,44 @@ def init_attr(self):
...
+            place = core.CUDAPlace(0)
+            if core.is_float16_supported(place) and check_grad:
+                self.check_grad_with_place(
+                    place, ['X', 'Alpha'],
+                    'Out',
+                    max_relative_error=max_relative_error)
+
+    cls_name = "{0}_{1}".format(parent.__name__, "Fp16Op")
+    TestPReluFp16Case.__name__ = cls_name
+    globals()[cls_name] = TestPReluFp16Case
```
Comment (thisjiang) [inline python/paddle/fluid/tests/unittests/test_prelu_op.py:304]: 原来python还有这种神奇写法😲学到了，这在C++中无法想象

**Review by thisjiang — APPROVED** [review]
LGTM

**Review by wanghuancoder — APPROVED** [review]
LGTM for @unittest.skipIf

**Review by Xreki — APPROVED** [review]
LGTM for the change of fp16's atol

**Pull request merged**